### ISSUE TITLE
autohidesScrollers seems to have no effect

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#3552]
I've just been trying to figure out how to get TXModular to display a horizontal scrollbar all the time, and the problem seems to be that (at least on my system) it is no longer possible to disable the behavior of automatically hiding scrollbars.

```
(
w = Window("fml", Rect(800, 200, 500, 400)).front;
v = ScrollView(w, w.view.bounds.insetBy(5, 5)).background_(Color.gray(0.5));
StaticText(v, Rect(400, 350, 200, 200)).background_(Color.blue(0.2));
v.autohidesScrollers = false;
)
```

Expected: Both scrollbars will be visible.

Actual behavior: Scrollbars are hidden. In fact, if I comment out `v.autohidesScrollers = false;`, the behavior in the window is exactly the same. `autohidesScrollers` seems to be completely ignored.

Moreover, at least on my system, the scrollbars don't appear when the mouse pointer touches the right or bottom borders. The only way to expose the scrollbars is to use the scroll wheel on the mouse (or two-finger scrolling on a touchpad). If you can manage to press and hold the mouse button while the stroller is visible, then you can grab it and it works okay. But that means the procedure to scroll is:

1. Position the mouse where you expect the scrollbar to appear.
2. Use the scroll wheel.
3. Quickly, before the scrollbar disappears, hold the mouse.

Wow. We could make it necessary to juggle chainsaws as well, but that might be overkill.

Any workarounds would be appreciated. I was considering using TXModular with students, but, if window scrolling is going to be difficult, I might have to use something else.

**Comment by jamshark70** [comment]
I discovered a really dirty hack -- change ProxyStyle.cpp to have:

```
static bool AlwaysShowScrollbars() {
// #if defined(Q_OS_MAC)
//   return QtCollider::Mac::AlwaysShowScrollbars();
// 
// #elif defined(Q_OS_X11)
//   return !QcApplication::SystemHasMouseWheel();
// 
// #elif defined(Q_OS_WIN)
//   return !QcApplication::SystemHasMouseWheel();
// 
// #else
//   return !QcApplication::SystemHasMouseWheel();
// #endif
  return true;
};
```

But if I'm going to use TXMod for teaching, I need a build like this for Mac and Windows.

Quite frustrating issue... everything works great except for one little oversight in... a scrollview? Really? Thousands of lines of Paul Miller's code brought to its knees by this.

Can someone at least confirm this?

**Comment by jamshark70** [comment]
Another note, posted to scdev but maybe useful to keep here:

Line 49 of ProxyStyle.cpp displays scroll bars if the control type is QStyle::CC_ScrollBar && AlwaysShowScrollBars() -- I wonder, is this perhaps overriding the QScrollArea's scroll bar policy properties? Shouldn't it be `&& (AlwaysShowScrollBars() || (something to check the scroll view's properties))`?

**Comment by jamshark70** [comment]
Aha, just found it.

https://doc.qt.io/archives/qtjambi-4.5.2_01/com/trolltech/qt/core/Qt.ScrollBarPolicy.html

- ScrollBarAlwaysOff - QAbstractScrollArea never shows a scroll bar.
- ScrollBarAlwaysOn - QAbstractScrollArea always shows a scroll bar.
- ScrollBarAsNeeded - QAbstractScrollArea shows a scroll bar when the content is too large to fit and not otherwise

For autohide = false, our class library is setting policy 0, never show.

Expected semantic: "don't autohide" = "always show"

Actual semantic: "don't autohide" = "we don't have to hide them because we aren't showing them at all"

So the fix is in the class library. I'll submit a PR later today.

**Comment by jamshark70** [comment]
Oh... I spoke too soon.

I had tested the class library change by running the reproducer, but I didn't interact with the window. But, just now, I reverted the dirty hack, rebuilt, and I now find different behavior from the behavior I observed before I started looking into this problem -- which I can't explain.

OK. My C++ sources are now clean (3.9). I modified autohidesScrollers so that 'false' sets the properties to 1 instead of 0.

Starting with this:

```
(
w = Window("fml", Rect(800, 200, 500, 400)).front;
v = ScrollView(w, w.view.bounds.insetBy(5, 5)).background_(Color.gray(0.5));
StaticText(v, Rect(400, 350, 200, 200)).background_(Color.blue(0.2));
)
```

- Original behavior before I started messing around: No scrollbars displayed at all, but they would appear when using the scroll wheel.
- Current behavior: Scrollbars are displayed, and then auto-hide. That's OK, as expected.

```
(
w = Window("fml", Rect(800, 200, 500, 400)).front;
v = ScrollView(w, w.view.bounds.insetBy(5, 5)).background_(Color.gray(0.5));
StaticText(v, Rect(400, 350, 200, 200)).background_(Color.blue(0.2));
v.autohidesScrollers = false;
)
```

- Original and current behavior: *Exactly the same* as the previous example. Scrollbars appear briefly and then auto-hide.

Now, strangely, if I revert `autohidesScrollers_` so that 'false' sets 0 (always hide), then I get the same behavior. So it still appears to me that `autohidesScrollers` is ignored.

**Comment by jamshark70** [comment]
Further -- and this is even more bizarre:

- If I reboot the interpreter and rerun the example, then (currently) the scrollbars display and do not hide -- *and* it does this whether or not I include `v.autohidesScrollers = false;` or not. I can do this repeatedly, same result.

- BUT, if I put the mouse in the window and touch the scroll wheel, then the scrollbars autohide (again, doesn't matter if autohidesScrollers is false or true). And this behavior persists through library recompilation, BUT if I reboot the interpreter again, then the autohide behavior stops.

I added a debugging post into `AlwaysShowScrollbars()` and found that `QcApplication::SystemHasMouseWheel()` is returning 0 at first, until I use the mouse wheel, and then it starts returning 1.

Now, what is also strange about this is that I'm quite sure I didn't see this behavior before rebuilding today.

So, the problems:

- `AlwaysShowScrollbars()` (https://github.com/supercollider/supercollider/blob/develop/QtCollider/style/ProxyStyle.cpp#L16-L29) seems to assume that Qt will accurately report whether the system does or does not have a mouse wheel, but on my system, I can demonstrate conclusively that this is not a valid assumption. It looks like Qt doesn't detect the mouse wheel until the user actually touches the mouse wheel within one of the app's windows. So this logic is wrong and needs to be fixed.

- `autohidesScrollers` sets the properties `\horizontalScrollBarPolicy` and `\verticalScrollBarPolicy`. I have not been able to make any difference in the behavior whether these properties are set to 0, 1, or 2. You can verify this by running any of these examples and then:
  ```
  v.setProperty( \horizontalScrollBarPolicy, 0 );
  v.setProperty( \horizontalScrollBarPolicy, 1 );
  v.setProperty( \horizontalScrollBarPolicy, 2 );
  ```
  So we are doing something internally to block the application of this property. I suspect https://github.com/supercollider/supercollider/blob/develop/QtCollider/style/ProxyStyle.cpp#L48, where the decision to display or not is solely based on `AlwaysShowScrollbars()`.

**Comment by jamshark70** [comment]
And, one more finding:

In ProxyStyle.cpp, I commented out the invocation of `AlwaysShowScrollbars()`:

```
  if (ctrl == QStyle::CC_ScrollBar /*&& AlwaysShowScrollbars()*/) {
```

... and now I get:

```
(
w = Window("fml", Rect(800, 200, 500, 400)).front;
v = ScrollView(w, w.view.bounds.insetBy(5, 5)).background_(Color.gray(0.5));
StaticText(v, Rect(400, 350, 200, 200)).background_(Color.blue(0.2));
)

// ^^^ scrollbars are continuously visible, no hiding
// then...

v.setProperty( \horizontalScrollBarPolicy, 0 );

// ^^ I thought this was "always hide"
// But the scrollbar is still visible

v.setProperty( \horizontalScrollBarPolicy, 1 );

// ^^ I thought this was "always show"
// But the scrollbar disappears!

v.setProperty( \horizontalScrollBarPolicy, 2 );

// ^^ I thought this was "as needed"
// Scrollbar is visible
```

Oh dear.

Qt 4.8 docs [1] | Qt 5 docs [2]
----------------|--------------
ScrollBarAlwaysOff | ScrollBar.AsNeeded
ScrollBarAlwaysOn  | ScrollBar.AlwaysOff
ScrollBarAsNeeded  | ScrollBar.AlwaysOn

[1] https://doc.qt.io/archives/qtjambi-4.5.2_01/com/trolltech/qt/core/Qt.ScrollBarPolicy.html
[2] https://doc.qt.io/qt-5/qml-qtquick-controls2-scrollbar.html#policy-prop

So, if sclang is built against Qt4, we need true -> 2, false -> 1, but against Qt5, we need true -> 0, false -> 2 :angry:

We might want to consider removing this scroll wheel logic. It looks like it was first implemented to deal with a WebKit bug (https://github.com/supercollider/supercollider/commit/480ea914a20523d4432f76ef0a46d986a1d7985c#diff-4c915152ab315a7f75b1c78d78b2ee62) but, based on my testing, it a/ doesn't work consistently and b/ breaks another feature (i.e., when I remove it, then the scrollbar policy property starts working as it should, but when it's present, that property takes no effect even when set directly).

Pinging @scztt (author of the aforementioned commit).

**Comment by scztt** [comment]
That change only affects WebViews, and I believe only in the IDE, FWIW.
From what I know, the scrollbar behavior should be like this:
- When the `view.autohidesScrollers == true`, a view will have scroll bars regardless of whether it needs them or not. If false, it will only have scroll bars in the directions where there's content to scroll. Note that this controls whether scroll bars *exist*, but doesn't affect the fade-out behavior.
- On Mac, there's a system-wide setting (Preferences -> General) that controls whether scrollbars are show **always**, **sometimes** (only when scrolling), or **never**. On Win and Linux (when I was last working on this, at least...), there were no such preferences. Instead, I connected the "fade-out" behavior to whether there was an X-Y scroll wheel present (usually a trackpad) - this is also not an easily queryable thing on Win/Linux, so I believe we simply assume there's not until we receive an actual X-Y scroll event.

These fixes were aligned towards fixing up behavior on Mac to conform with OS standards, and fixing Win and Linux to behave about the same (since there weren't OS-level norms I was aware of at the time). There are probably two ways to improve this for Linux and Win:

1. Connect `AlwaysShowScrollbars()` to OS-specific scroll behavior settings on Win and Linux, like it is on Mac. This is the best option, **if** these OS-level preferences exist.
2. Add a global e.g. "fadeScrollbars" setting to QtGUI, and patch it in to `AlwaysShowScrollbars()` in addition to the existing checks (e.g. `!fadeScrollbars ||  !QcApplication::SystemHasMouseWheel()`). This should definitely be set to "OS default" on Mac, it means sclang basically ignores the actual user preferences for scrollbar behavior, which is pretty disruptive and annoying.

I agree that `autohidesScrollers` is a bit confusing in this context - it's grandmothered in from previous versions of SC, we should probably rename and deprecate.

**Comment by jamshark70** [comment]
> This should definitely be set to the OS default on Mac - else it means sclang basically ignores the actual user preferences for scrollbar behavior, which is pretty disruptive and annoying.

Sure.

I'd be fine with it if the Linux/Win ifdef branches simply return true, and keep the OS preference check for OSX. Also we need to be certain in OSX that the scrollbars reappear when mousing over. I get it -- fade-out is a lovely idea. But it has to *work*. If it doesn't work, then, dump it.

Connecting the fade behavior to the scroll wheel is bad for user experience in Linux.

**Comment by jamshark70** [comment]
> Add a global e.g. "fadeScrollbars" setting to QtGUI, and patch it in to AlwaysShowScrollbars() in addition to the existing checks (e.g. !fadeScrollbars || !QcApplication::SystemHasMouseWheel()). This should definitely be set to the OS default on Mac - else it means sclang basically ignores the actual user preferences for scrollbar behavior, which is pretty disruptive and annoying.

I'm willing to have a crack at this. And I agree, a user preference is a good idea.

Would this variable will be better as a global, or a (static?) member variable of ProxyStyle? Somehow a member variable seems cleaner -- especially if this class is also used in the IDE for the help browser.

But I'm not sure where/how ProxyStyle is used. A hypothetical `QtGui.fadeScrollbarPolicy_()` method would have to delegate to a primitive, and that primitive would have to access instances of ProxyStyle. I'm not clear how to do that.

Maybe I don't know enough C++ to do it right, but if it's going to take time for someone else to pick it up, I'll give it a shot with some guidance.

**Comment by jamshark70** [comment]
Anyone?

I confirmed that Windows has the same behavior as Linux wrt auto-fading the scrollers. And I confirmed that at least one of my students is on Windows. Once the TXMod patches grow beyond a minimal level of complexity, it'll get painful.

Kind of a perfect storm here... a patch awhile back that was supposed to affect only webviews but which actually affects ScrollViews as well, an app that depends on horizontal scrolling, and some really unfortunate mishandling by Qt of faded-out scrollbars.

I'm fine with mac reverting to the system preference, but that has-scroll-wheel logic for Linux and Windows is burdened by unintended consequences and it's going to cause a problem for my teaching. I don't want to be a pest but I need a workaround for this.

**Comment by brianlheim** [comment]
> I don't want to be a pest but I need a workaround for this.

Sorry, but reading this it's kind of hard for me to follow what exactly the problem is. Can you summarize what needs to be done, and I'll look into it in the next week or two if nobody else has.

**Comment by joshpar** [comment]
And I’m also happy to help (tonight and tomorrow), but I need to know what the steps are at this point that need to be corrected.
Josh

**Comment by scztt** [comment]
> That's reversed, isn't it? By this description, autohide = true means never hide.
You're right - sorry, a typo.

> Sorry to report, but the effect is most definitely not contained to the IDE.
The change in [editors/sc-ide/widgets/style/style.cpp](https://github.com/supercollider/supercollider/commit/480ea914a20523d4432f76ef0a46d986a1d7985c#diff-b7f76147b487e923bba074a62df9dbf2L78) only affects WebView's in the IDE. I think this workaround may be removed in the QT 5.9 branch, as I believe this underlying QT bug was fixed.

This should probably be fixed by a classvar in QtGUI, backed by a get/set primitive, that works *something* like:

    QtGUI.sliderFadeBehavior = \auto; // current behavior - use system pref if available, else autohide only when there's an X-Y scroll wheel present
    QtGUI.sliderFadeBehavior = nil; // equivalent to \auto
    QtGUI.sliderFadeBehavior = \noFade;
    QtGUI.sliderFadeBehavior = \fade; // fade out scrollers, regardless of system pref or presence of XY scroller

This should be documented as something to be set in a users startup file, and **NEVER EVER** by a Quark or via a classlib operation.

**Comment by scztt** [comment]
It would be good to do some research to figure out if there's a system-wide preference on linux or on windows that controls this behavior across the OS - we should really follow this if it's available. I would be surprised if there isn't one on linux - linux has prefs for everything :)

It's also worth a look through the QT documentation to see if there are settings regarding the mouse-over tolerances of scroll bars - it sounds like the true misbehavior here is just that the tolerances on some platforms make it hard to reveal scrollers using mouseover.

**Comment by jamshark70** [comment]
> The change in editors/sc-ide/widgets/style/style.cpp only affects WebView's in the IDE.

But the fact is, when I change the code here, ScrollViews in sclang behave differently. I understand that it wasn't supposed to affect anything else, but my observation is that it does.

This wouldn't be the first time a piece of code had an effect outside of the place where it was supposed to.

**Comment by jamshark70** [comment]
> Sorry, but reading this it's kind of hard for me to follow what exactly the problem is. Can you summarize what needs to be done, and I'll look into it in the next week or two if nobody else has.

Short term, I need a Windows build that replaces `return !QcApplication::SystemHasMouseWheel();` with `return true;` in ProxyStyle.cpp, function `static bool AlwaysShowScrollbars()`. That's not meant for release, just for a couple of students (who need to start doing homework with txmod next week).

(If I submit a PR, can I get an installer from AppVeyor? I don't know if it does every PR or only commits merged into release branches.)

(Or, how many hours should I expect to spend to prepare a Windows build environment?)

The proper fix is as Scott C. outlines:

- Look for system preferences about auto-fading scrollbars and use those instead of the scroll wheel check. Or, failing that, add a preference-setting method in sclang.

- Figure out why Qt isn't un-fading them in some platforms. If we can't figure this out, then we might have no choice but to override system preferences.

**Comment by jamshark70** [comment]
> The change in editors/sc-ide/widgets/style/style.cpp

PS Scott -- I'm talking about ProxyStyle.cpp, not style.cpp.

**Comment by brianlheim** [comment]
> If I submit a PR, can I get an installer from AppVeyor?

It builds branches made on the main repo. If you make a branch here, you can get a zip. Not installer, that's only for tagged commits e.g. releases. Message me if you have any issues with that.

**Comment by jamshark70** [comment]
> It builds branches made on the main repo. If you make a branch here, you can get a zip.

Aha, ok, I can handle that by myself. That might handle the immediate need (since my students on Mac can just change the system preference). Thanks.

**Comment by scztt** [comment]
@jamshark70 For the future - if you'd like to set up your personal GH repository so that you can produce and post travis (and probably Appveyor) builds, email me offline and I can send you instructions.

**Comment by jamshark70** [comment]
> Figure out why Qt isn't un-fading them in some platforms.

I've now checked scrollbar behavior in Windows and Mac as well, and as far as I can see, it appears to be intentional behavior on Qt's part. Once the scrollbars fade away, the only way to bring them back is to use the mouse wheel or two-fingered scrolling.

If there's any hidden scrollbar area where mousing over causes scrollbars to reappear, I can't find it, on any platform.

When scrollbars are *not* fading away, the scroll area shrinks and is replaced by a translucent gray bar. Mousing over this bar causes a full scrollbar to reappear.

This behavior should definitely not be forced upon users.

I think we should move ahead (quickly) with a user preference. My questions are still open. https://github.com/supercollider/supercollider/issues/3552#issuecomment-370218438

**Comment by scztt** [comment]
I think I answered your question re API  (or, at least, stated my preferences :) ) [here](https://github.com/supercollider/supercollider/issues/3552#issuecomment-370918063)?

`_QtGUI_DebugLevel`, `_QtGUI_SetDebugLevel` should be a good start starting point for how to implement it as a primitive.

For clarity, on Mac at least (the only platform I have immediate access):
1. This is not a QT behavior, this is the OS-wide scrollbar behavior.
2. It's not forced - there's a user preference that sets the scrollbar fading behavior, so users can set it to however they prefer.

**Comment by jamshark70** [comment]
Interesting. I don't have a Mac system to check, but right now, I'm looking at both an xfce4 desktop on my Linux machine and a Windows 7 desktop on my school machine. Both are showing scrollbars all the time.

In xfce4, I get standard scrollbars in Firefox, the terminal, the file browser etc., and also in sc-ide. But in sclang GUIs, I get this:

![scrollers](https://user-images.githubusercontent.com/318301/37498403-2a15f1e0-28f9-11e8-995e-01710078dca1.png)

And if I mouse over the gray bar:

![scrollers2](https://user-images.githubusercontent.com/318301/37498414-3233be7a-28f9-11e8-9089-26ffde394e1f.png)

The Unity desktop (for Ubuntu) has similar collapsing scrollbars -- https://askubuntu.com/a/121501/571420 -- note that this illustration comes from an askubuntu thread about how to disable collapsing! So I assume this is not a universally popular feature.

FWIW I think Apple made a mistake to set this option by default, and Qt made a mistake to copy OSX's behavior.

I could also make a case for SC overriding the system behavior. My reason: Scroll wheels and trackpad scrolling make another UI mistake. If you point the mouse and a slider and use the scroll wheel, it will not scroll the window, but instead change the value of the slider. Now, if you're scrolling window contents and accidentally this causes a slider or number box to pass underneath the (stationary) mouse pointer, in every implementation I've seen, magically the window stops scrolling and it starts changing the value. This almost never happens in web browsers, but it happens *all the time* in control heavy windows like DAWs or sclang performance interfaces. I can't count the number of times I was scrolling in Cubase and accidentally hooked onto the value box for an automation parameter and -- poof! Mix ruined. It's an incredibly stupid interface design blunder.

And here's why it's such a problem specifically for TX Modular (screenshot below). Control channels have empty space above the fader. Audio channels have sends. So, you put the mouse pointer in the empty space, start scrolling horizontally, and *bang* -- the second you scroll past an audio channel, you're no longer scrolling *and* you're changing a send level. *And*, because of Apple's arrogance when it comes to interfaces, the *only* way to scroll this panel (with default system settings) is using the scroll wheel, so it's very very easy to crash into this problem.

I know I'm running on too long, and I'm sorry about that, but to be honest, I'm not satisfied with the answer "this is a system preference and we should respect it." The system preference is designed for web browsers. SC is not a web browser. Its GUIs have different types of controls, which happen to interact badly with scroll wheels. SC should reduce dependency on scroll wheels.

I'll look into the implementation when I have time. I think we should consider giving the user's SC preference priority over the system preference for Mac.

![txmod-mixer](https://user-images.githubusercontent.com/318301/37498429-44221ef6-28f9-11e8-8350-9fb7b59a6f83.png)

**Comment by jamshark70** [comment]
Try it yourself. Good luck scrolling this vertically with the scroll wheel.

```
(
w = Window("scroll me", Rect(800, 200, 500, 400));
v = ScrollView(w, w.view.bounds.insetBy(5, 5));
a = Array.fill(30, { |i|
	Slider(v, Rect(5, 5 + (i * 20), v.bounds.width - 10, 17))
	.orientation_(\horizontal)
	.value_(i / 29);
});
w.front;
)
```

**Comment by scztt** [comment]
I think the example of the scrollable view filled with scroll-capable controls convinced me that it would be valuable to have auto-fade behavior controllable on a per-View level of granularity.

Your example works fine on my mac - it just scrolls, and doesn't affect the sliders. You can sort of get the sliders to respond to a scroll event if there's no more room to scroll and you have the cursor in the right place, but this feels more like a Qt bug given that this works reasonably in all other cases.

**Comment by telephon** [comment]
>Your example works fine on my mac - it just scrolls, and doesn't affect the sliders. 

Not for me here. I get the same behavior that james describes on macOS 0.11.6.

**Comment by jamshark70** [comment]
> I think the example of the scrollable view filled with scroll-capable controls convinced me that it would be valuable to have auto-fade behavior controllable on a per-View level of granularity.

I think I'd like that even better. But it's a different implementation, and I'm not sure where the pieces fit together.

So I guess we would need to add a member variable to QcScrollArea (or maybe two, if we want to disable fadeout independently for horizontal and vertical scrollbars). But the fade behavior is set in `ProxyStyle::drawComplexControl()` -- does this function have access to the QcScrollArea object? All I know currently is that the argument `const QWidget * w` is nonzero when it's invoked for ScrollView components, but which object it's pointing to, I have no idea.

If we want independent control over H and V scrollbars, then `ProxyStyle::drawComplexControl()` needs to distinguish between them -- I guess http://doc.qt.io/qt-5/qstyleoptionslider.html#orientation-var.

**Comment by jamshark70** [comment]
Retested with Qt 5.11. Same behavior. Scrollbars do NOT automatically reappear when you mouse over them.

**Comment by gusano** [comment]
> Your example works fine on my mac - it just scrolls, and doesn't affect the sliders. You can sort of get the sliders to respond to a scroll event if there's no more room to scroll and you have the cursor in the right place, but this feels more like a Qt bug given that this works reasonably in all other cases.

It also works fine for me (archlinux, Qt-5.11)

**Comment by jamshark70** [comment]
I was literally just two days ago scrolling over some sliders and getting stuck with sliders' values changing.

I have no idea why this is so different on various systems, but I think it's relevant only insofar as users who aren't experiencing the problem can't test any fixes (or if it leads to identifying some settings that make a difference). Otherwise, I don't see that it has much bearing.

**Comment by snappizz** [comment]
after #3686, how serious is this issue? i have it marked as critical for 3.10 but i think it can probably safely be delisted.

**Comment by jamshark70** [comment]
I think, under #3686, there was at least one comment to the effect that the user didn't want to have scrollbars displayed all the time on his system, just because it doesn't work properly on some other systems.

That's a fair comment. Whether that makes it critical for 3.10 or not, I don't know.

Before #3686, we had less clutter in ScrollViews, acceptable behavior in some systems, and severely unusable behavior in others (and, unintentionally, the burden was basically completely on me to try to figure out how to have usable behavior + fading scrollers). Now we have usable behavior in all systems at the cost of a bit more visual clutter in Linux and Windows, and more of the burden is shared. I would like to see fading scrollers, in fact -- but now there might be more interested parties to work on it.

**Comment by gusano** [comment]
@snappizz @jamshark70 I believe that I was complaining about the fact that now scrollbars always display no matter what (while I couldn't reproduce the original scrolling issue). Although it makes less *slick* UI design I wouldn't make it critical for 3.10.

ps: let me know if I should open another issue about that so we can safely close this one

**Comment by mikeuwins** [comment]
I see I am a long time after this thread was opened - but is it still impossible to hide / disable scrollbars in ScrollView for Linux?  I am desperate to get rid of the scroll bars on my GUI and after half a day completely bamboozled, I found this thread and wondered if, 7 years on - it's still a thing.  If they have been permanently enabled, is there a workaround so I can disable them?

Thanks,
Mike
